gramex/services/storelocations.py

```diff
@@ -0,0 +1,117 @@
+import gramex.data
+import json
+import time
+from uuid import uuid4
+from typing import Union
+
+
+class StoreLocation(object):
+    def __init__(self, url: str, table: str, columns: dict = None, **kwargs: dict):
+        '''Set up a data store. Used by gramex.yaml `storelocations:` service.
+
+        Examples:
+            >>> store = StoreLocation(url='sqlite:///x.db', table='x', columns={'y': 'text'})
+            >>> store.insert(y='hello')
+
+        Parameters:
+            url: sqlalchemy URL
+            table: table name
+            columns: column names, with values are SQL types, or dicts
+            **kwargs: passed to `sqlalchemy.create_engine()`.
+
+        `columns` can be SQL type strings (e.g. `"REAL"` or `"VARCHAR(10)"`) or a dict with keys:
+
+        - `type` (str), e.g. `"VARCHAR(10)"`
+        - `default` (str/int/float/bool), e.g. `"none@example.org"`
+        - `nullable` (bool), e.g. `False`
+        - `primary_key` (bool), e.g. `True` -- used only when creating new tables
+        - `autoincrement` (bool), e.g. `True` -- used only when creating new tables
+
+        The parameters are the same as [gramex.data.alter][].
+        '''
+        self.store = {'url': url, 'table': table, 'columns': columns, **kwargs}
```

Comment:
Need to document that the store structure has to be adhered to irrespective of what storage location is used.

Reply:
Documented this.